vertical-pod-autoscaler/docs/proposals/crash_loop_handling/README.md

```diff
@@ -0,0 +1,196 @@
+- Resource requests metrics for CPU and memory are obtained for each container in a pod from the prometheus client
+- The history obtained is initialized with the recommender to start estimating the target, lower and upper bounds for the containers, and there by the pods
+- Target Estimator, Lower Bound Requests and Upper Bound Requests are the three main estimations we determine using recommender
+- A percentile value for each of these estimators are set as ![follows](https://github.com/kallurbsk/autoscaler/blob/b46a54192da8ab4181346beba13ae1579db82063/vertical-pod-autoscaler/pkg/recommender/logic/recommender.go#L101)
+  - 0.9 percentile for target CPU and Memory
+  - 0.5 for lower bound requests of CPU and Memory
+  - 0.95 for upper bound requests of CPU and Memory
+- For Target Estimator we only recommend Percentile Estimation with a safety margin of 15% by default
+- An exponential variation in the lower bound requests, upper bound requests which is determined by history of metrics monitored and pod life span. There is a "confidence" value on which the entire resource scaling is determined.
+- For lower bound and upper bound estimation a configurable safety margin (by default 15%) is applied
+- The current lower bound and upper bound estimation formula is as follows
+```
+scaledResource = originalResource * (1 + 1/confidence)^exponent
+
+confidence = (LastSampleTimeStart - FirstSampleTimeStart) / (no. of hours per day i.e 24)
+```
+
+Where,
+- **scaledResource**  : newly scaled integer value of the amount of the 
+                    resource  (CPU or Memory)
+- **originalResource**: current integer value of the amount of the
+                    resource (CPU or Memory)
+- **confidence**      : An non-negative integer number that heuristically measures 
+                    how much confidence the history aggregated in the AggregateContainerState provides. For a workload producing a steady stream of samples over N days at the rate of 1 sample per minute, this metric is equal to N. It is calcualted as the total count of samples and the time between the first and the last sample 
+- **exponent**        : A non-zero float exponent value which is determined by the 
+                    targetEstimator, lowedBoundEstimator or upperBoundEstimator. Each of these estimators are set through percentile estimations. targetEstimator is set to percentileEstimator while lowerBoundEstimator and upperBoundEstimator are set as confidence multipliers. For lowerBoundEstimator the exponent is set to -2 and for upperBoundEstimator it is set to +1
+
+## Proposal
+
+To address the concerns mentioned we are proposing this change in the recommendation engine of VPA recommender for autoscaler component.
+
+#### Scale Up:
+Doubling of resources based on current resource usage
+  1. Today we depend on the resource requests history from prometheus client for a history length of 8 days (default)
```

Comment:
Hmm, I think in gardener we only use the metrics-server for vpa recommender (please correct me if I'm wrong).
Does it make any difference, if the metrics-server is used for feeding the recommender initialization compared to `metrics-server` + `VPACheckpoint`?

Reply:
Metrics server, VPA checkpoints, Prometheus for getting usage. 
For the second question, I am not entirely sure.